plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/common/core/utils/UrlUtils.java

```diff
@@ -163,25 +167,28 @@ public static boolean hasScheme(String host) {
 		if (isEmpty(host)) {
 			return false;
 		}
-		return host.indexOf(SCHEME_SEPARATOR) > -1;
+		return host.contains(SCHEME_SEPARATOR);
 	}
 
 	public static String getHost(String url) {
 		if (isEmpty(url)) {
 			return url;
 		}
-		
-		int beginIndex = getHostIndex(url);
-		if (beginIndex == -1) {
-			beginIndex = 0;
+		String host = null;
+		if (url.contains(SCHEME_SEPARATOR)) {
+			try {
+				host = new URI(url).getHost();
+			} catch (URISyntaxException ignored) {
+			}
 		}
-		
-		int endIndex = url.length() - 1;
-		if (url.endsWith("/")) {
-			endIndex -= 1;
+
+		if (host == null) {
+			Matcher m = SIMPLE_QUASI_URL_PATTERN.matcher(url);
```

Comment:
what's the difference btw. SIMPLE_QUASI_URL_PATTERN and SIMPLE_URL_PATTERN? Cant we reuse?